src/player_activity.h

```diff
@@ -44,51 +45,97 @@ enum activity_type {    // expanded this enum for readability
 
 class player_activity : public JsonSerializer, public JsonDeserializer
 {
-    public:
-        activity_type type;
-        int moves_left;
-        int index;
-        int position;
-        std::string name;
-        bool ignore_trivial;
-        std::vector<int> values;
-        std::vector<std::string> str_values;
-        point placement;
-        bool warned_of_proximity; // True if player has been warned of dangerously close monsters
-        // Property that makes the activity resume if the previous activity completes.
-        bool auto_resume;
+  void burrow_do_turn(player*);
```

Comment:
I don't think these need to be ing the player_activity class, other than making their access to member variables be slightly more streamlined, there's no reason for it.  They're really just misc chunks of code, and they should probably operate at arm's length from the player activity class to respect it's interface.
Also they don't need to be in the interface at all, they can be static functions that are called by some player activity dispatchers (currently do_turn() and finish())